lib/src/components/NxDropdown/NxDropdown.tsx

```diff
@@ -50,28 +45,37 @@ const NxDropdown: FunctionComponent<Props> = function NxDropdown(props) {
       child
   ));
 
-  const toggle = (
-    <NxButton ref={toggleRef}
-              type="button"
-              variant={variant || 'tertiary'}
-              className={buttonClasses}
-              onClick={!disabled && onToggleCollapse || undefined}
-              aria-haspopup="true"
-              aria-expanded={isOpen}>
-      <span className="nx-dropdown__toggle-label">{ label }</span>
-      <NxFontAwesomeIcon icon={isOpen ? faCaretUp : faCaretDown} size="lg" />
-    </NxButton>
-  );
+  const toggleElement: AbstractDropdownToggleElement = ({ onToggleCollapse, toggleRef }) => {
+    const button = (
+      <NxButton ref={toggleRef}
+                type="button"
+                variant={variant || 'tertiary'}
+                className={buttonClasses}
+                onClick={!disabled && onToggleCollapse || undefined}
+                aria-haspopup="true"
+                aria-expanded={isOpen}>
+        <span className="nx-dropdown__toggle-label">{ label }</span>
+        <NxFontAwesomeIcon icon={isOpen ? faCaretUp : faCaretDown} size="lg" />
+      </NxButton>
+    );
+
+    return toggleTooltipProps ? <NxTooltip { ...toggleTooltipProps } >{button}</NxTooltip> : button;
+  };
 
   return (
-    <div className={classes} onKeyDown={onKeyDown} {...attrs}>
-      { toggleTooltipProps ? <NxTooltip { ...toggleTooltipProps } >{toggle}</NxTooltip> : toggle }
-      { isOpen && <NxDropdownMenu ref={menuRef} onClosing={onMenuClosing}>{wrappedChildren}</NxDropdownMenu> }
-    </div>
+    <AbstractDropdown className={classes}
+                      isOpen={isOpen}
+                      disabled={disabled}
+                      toggleElement={toggleElement}
```

Comment:
I think I'd rather this prop on `AbstractDropdown` be named `renderToggleElement`, since its type is a function which takes some properties from the `AbstractDropdown` and returns the actual (JSX) element

Reply:
> I think I'd rather this prop on `AbstractDropdown` be named `renderToggleElement`, since its type is a function which takes some properties from the `AbstractDropdown` and returns the actual (JSX) element

Renamed it here: d080533d2c8381e03a063ff766639f8ef30c2d80 :)